planner/funcdep/fd_graph.go

```diff
@@ -727,10 +814,17 @@ func (s *FDSet) MakeOuterJoin(innerFDs, filterFDs *FDSet, outerCols, innerCols F
 	for _, edge := range innerFDs.fdEdges {
 		// Rule #2.2, constant FD are removed from right side of left join.
 		if edge.isConstant() {
+			// for constant FD
+			// 1: column = definite value; add cond-fd waiting for null-reject. (not in not-null columns from inner cols and join predicate)
+			// 2: column is null; add constant fd directly.
+			constantNull := edge.to.Difference(innerFDs.NotNullCols.Union(filterFDs.NotNullCols))
+			s.AddConstants(constantNull)
+			s.AddNCFunctionalDependency(edge.from, edge.to.Difference(constantNull), innerCols, edge.strict, edge.equiv)
```

Comment:
When would we need this case?

Reply:
```
// Constant FD:
// In our implementation, constant FD is a special kind of strict functional dependency.
// When discuss the inference rule of constant FD, we can utilize some rule from strict functional dependency above, adding some special consideration.
// 1: Any constant functional dependency f: {} -> {y} that held in S continues to hold in Q.
// 2: Any constant functional dependency f: {} -> {y} that held in T may be decomposed to multi strict FD first:
//       1: if y1 ⊆ y and y1 are null constant (come from y1 is null), then constant FD {} -> {y1} saved.
//       2: for other y, a cond-fd are preserved with null-constraint cols as T.
// 3: Any constant function dependency f: {} -> {y} that produced from predicate P, decomposing to multi strict FD first:
//       1: for part of y ⊆ T, do it like what 2 says.
//       2: for part of y ⊆ S, do it like a cond-fd with null-constraint cols as T.
```
here you comment means，some constant FD on the left join inner side will be stored as NCEdge, here corresponding to point 2.2.

so the case you can found from the test I've marked is:
```
```